Jenkinsfile

```diff
@@ -69,6 +69,21 @@ pipeline {
         }
       }// END steps
     } // END stage
+
+    stage('Trigger Snapshot Deployment') {
+      when {
+        allOf {
+          expression {
+            currentBuild.getBuildCauses().toString().contains('BranchIndexingCause')
```

Comment:
It happena when the job is started by "branch indexing" (multi-branch pipeline job).

I tried to run the job manually, and the stage is not executed because the condition does not hold.

When a job is triggered by schedule, the build cause is again different. I haven't tested this case, but the condition should not hold.

The cause of the build is shown at the very beginning of the build log.

That's all I can say, and tested :)

Reply:
so also manually calling scan multi branch pipeline etc wont triger it.

Reply:
from what I know, calling a multi branch pipeline scan does not necessarily start jobs: it only starts jobs for new branches (but this change only concerns the main branch) or for branches that have new git commits.